splight_lib/models/hub.py

```diff
@@ -147,15 +147,36 @@ def delete(self):
 
     def download(self):
         hub_client = get_hub_client()
-        params = {"name": self.name, "version": self.version}
-        return hub_client.download(data=params)
+        return hub_client.download(id=self.id, name=self.name, type="source")
+
+    def save(self):
+        hub_client = get_hub_client()
+        return hub_client.save(instance=self)
```

Comment:
```suggestion
        return hub_client.save(instance=self.model_dump())
```
Check the client always get the model in the dict format. That is the approach used in the public library